examples/relay/main.go

```diff
@@ -91,21 +125,30 @@ func run() {
 	// prevent us from redialing again so quickly. Since we know what we're doing, we
 	// can use this ugly hack (it's on our TODO list to make it a little cleaner)
 	// to tell the dialer "no, its okay, let's try this again"
-	h1.Network().(*swarm.Swarm).Backoff().Clear(h3.ID())
+	unreachable1.Network().(*swarm.Swarm).Backoff().Clear(unreachable2.ID())
+
+	log.Println("Now let's attempt to connect the hosts via the relay node")
 
-	h3relayInfo := peer.AddrInfo{
-		ID:    h3.ID(),
+	// Open a connection to the previously unreachable host via the relay address
+	unreachable2relayinfo := peer.AddrInfo{
+		ID:    unreachable2.ID(),
 		Addrs: []ma.Multiaddr{relayaddr},
 	}
-	if err := h1.Connect(context.Background(), h3relayInfo); err != nil {
-		log.Printf("Failed to connect h1 and h3: %v", err)
+	if err := unreachable1.Connect(context.Background(), unreachable2relayinfo); err != nil {
+		log.Printf("Unexpected error here. Failed to connect unreachable1 and unreachable2: %v", err)
 		return
 	}
 
+	log.Println("Yep, that worked!")
+
+	// **** MIGHT BE WORTH ADDING A COMMENT ADDRESSING WHAT WithUseTransient() is here,
+	// and why it is needed ******************
```

Comment:
Can you do that? :)

Reply:
I'm not really confident with transient connections, as I'm new to libp2p 😅. From what I can piece together from reading some source comments is that transient stream are those that have opened a connection but the stream is not yet fully established, as a protocol hasn't been negotiated yet. 

So, my guess would be that, in the context of a circuit relay, we have established an initial connection to the relay peer (which is marked as transient) but we don't have a connection to our destination peer yet, so when we open our stream to the destination peer we indicate that we are using a transient connection, but it is isn't a problem as the relay will connect to the destination peer and our connection will be upgraded once we've negotiated our stream with the destination peer. But it is possible I'm way off there 😁

Reply:
> From what I can piece together from reading some source comments is that transient stream are those that have opened a connection but the stream is not yet fully established, as a protocol hasn't been negotiated yet.

Yeah, confusing terminology here. Sorry for that. That would be transient in the context of the resource manager. In this context, it means something completely different.

Transient here means that it's a connection established on a relayed (as opposed to a direct) connection. Since the relay limits the amount of data that can be exchanged over the relayed connection, application controls need to explicitly opt-in into using a relayed connection. In general, they should only do that if they have low bandwidth requirements, and they're ok with the connection being killed when the relayed connection is replaced with a direct (holepunched) connection.

I hope this explanation makes sense.

Reply:
Ah that's interesting, thanks for the details. Is there anything in the docs or source I could read to explore this a bit more? For example, how an application/relay node would go about lifting the relay data limits, or upgrading the connection for devices where hole punching is not possible.